python/pyspark/sql/dataframe.py

```diff
@@ -3175,12 +3177,12 @@ def melt(
 
         Parameters
         ----------
-        ids : str, Column, tuple, list, optional
-            Column(s) to use as identifiers. Can be a single column or column name,
-            or a list or tuple for multiple columns.
-        values : str, Column, tuple, list, optional
-            Column(s) to unpivot. Can be a single column or column name, or a list or tuple
-            for multiple columns. If not specified or empty, uses all columns that
+        ids : str, list, tuple, optional
+            Column name(s) to use as identifiers. Can be a single column name,
+            or a list or tuple for multiple column names.
+        values : str, list, tuple, optional
```

Comment:
Seems like `values` shouldn't be described as an optional, as it is a required argument.  How about something like this?

```python
        values : str, Column, tuple, list or None
            Column(s) to unpivot. Can be a single column or column name, or a list or tuple
            for multiple columns. If specified, and not `None` must not be empty. If `None` uses all
            columns that are not set as `ids`.
```

Reply:
We could also provide defaults  for this and the following arguments:

```python
def unpivot(
        self,
        ids: Optional[Union[str, List[str], Tuple[str, ...]]],
        values: Optional[Union[str, List[str], Tuple[str, ...]]] = None,
        variableColumnName: str = "variable", 
        valueColumnName: str = "value",
) -> "DataFrame":
```